### PR TITLE
updated Mongoid to v. 6

### PR DESCRIPTION
**Pull request by moodlemags**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ## 5.3.1 (Next)
 
+* Upgraded mongoid to 6.0.0 - [@moodlemags](https://github.com/moodlemags).
```
Comment (dblock) [inline CHANGELOG.md]: Change this to `[#231](...): Compatibility with Mongoid 6 - [...]()`.

**Comment by dblock** [comment]
Looks good. Amend the changelog with the pull request number please and try to squash your commits (good exercise).

**Comment by dblock** [comment]
You still need to add mongoid 6 to https://github.com/mongoid/mongoid-slug/blob/master/.travis.yml, which is the CI.

**Comment by dblock** [comment]
Bump @moodlemags. See my comment above.

**Comment by dblock** [comment]
Squash your commits as well to make this a single commit with a nice the description.

**Comment by Fudoshiki** [comment]
Hi, @dblock you can squash PR commits on accepting it with squash commits option

**Comment by dblock** [comment]
@Fudoshiki Thanks. I was helping @moodlemags IRL make her first (second) open-source PR, while github lets you squash commits it's good practice.

Anyone wants to finish this?

**Comment by moodlemags** [comment]
@dblock Oh my gosh! I got so focused on the other one this one slipped past. I am on it!

**Comment by moodlemags** [comment]
@dblock umm so I seem to have accidentally rebased everything and have been trying to undo it and cannot. Is it possible? I'm considering making just a new branch and doing from that. Eek!

**Comment by dblock** [comment]
New branch can work too, feel free to close this and reopen a new PR. Or you can force push over your other branch.

**Comment by dblock** [comment]
Alright! Now the build is failing for a legit reason, see https://travis-ci.org/mongoid/mongoid-slug/jobs/168612055.

**Comment by moodlemags** [comment]
it is sneaky! I have found where just figuring out the how to fix now

**Comment by dblock** [comment]
It's probably missing the paranoia gem for mongoid 6. It used to be part of mongoid and was moved into being an external dependency around version 4.

File: Gemfile [hunk 2]
```diff
@@ -2,12 +2,14 @@ source 'https://rubygems.org'
...
 
-case version = ENV['MONGOID_VERSION'] || '5'
+case version = ENV['MONGOID_VERSION'] || '6'
 when 'HEAD'
   gem 'mongoid', github: 'mongodb/mongoid'
   gem 'mongoid_paranoia', github: 'simi/mongoid_paranoia'
   gem 'mongoid-observers'
   gem 'rails-observers', github: 'rails/rails-observers'
+when /^6/
+  gem 'mongoid', '~> 6.0.0'
```
Comment (Fudoshiki) [inline Gemfile]: add this here
```ruby
gem 'mongoid_paranoia', github: 'simi/mongoid_paranoia'
gem 'mongoid-observers'
gem 'rails-observers', github: 'rails/rails-observers'
```

Comment (dblock) [inline Gemfile]: @Fudoshiki feel free to finish this up in a separate PR!

**Comment by marceloboeira** [comment]
Thanks for the nice work everyone!

What is missing to support mongoid6 ?

I am stuck on this - > https://github.com/mongoid/mongoid-slug/issues/233

I believe it is related, I wanted to check the tests with mongoid6 in order to try to open a PR, but there is no sense to attempt anything while this one is not ready.

**Comment by dblock** [comment]
@marceloboeira Just a passing build here! Feel free to take over.

**Comment by dblock** [comment]
Ignore me, just taking this over, https://github.com/mongoid/mongoid-slug/pull/234.

**Comment by dblock** [comment]
@marceloboeira notice that the gem as released works, there were no code changes

**Comment by marceloboeira** [comment]
@dblock yeap, I am using here, but I will sleep better knowing that the build is green for mongoid 6.x 👍

**Pull request closed without merging**